src/Kiota.Builder/Writers/Shell/ShellCodeMethodWriter.cs

```diff
@@ -622,10 +757,23 @@ private static void WriteRequestInformation(LanguageWriter writer, CodeMethod ge
                 writer.WriteLine($"if ({paramIdent} is not null) requestInfo.Headers.Add(\"{paramName}\", {paramIdent});");
             }
 
-            // Set the content type header. Will not add the code if the method has no RequestBodyContentType or if there's no body parameter.
-            if (generatorMethod.Parameters.Any(p => p.IsOfKind(CodeParameterKind.RequestBody)) && !string.IsNullOrWhiteSpace(generatorMethod.RequestBodyContentType))
+            // Set the content type header. Will not add the code if the method is a stream, has no RequestBodyContentType or if there's no body parameter.
+            if (!isStream && generatorMethod.Parameters.Any(p => p.IsOfKind(CodeParameterKind.RequestBody)))
             {
-                writer.WriteLine($"requestInfo.SetContentFromParsable({RequestAdapterParamName}, \"{generatorMethod.RequestBodyContentType}\", model);");
+                if (!string.IsNullOrWhiteSpace(generatorMethod.RequestBodyContentType))
+                {
+                    writer.WriteLine($"requestInfo.SetContentFromParsable({RequestAdapterParamName}, \"{generatorMethod.RequestBodyContentType}\", model);");
+                }
+                else
+                {
+                    // Being here implies a new case to handle.
+                    var url = generatorMethod.Parent is CodeClass c ? c.Properties.OfKind(CodePropertyKind.UrlTemplate)?.Name : null;
+                    if (string.IsNullOrWhiteSpace(url))
+                    {
+                        url = "N/A";
+                    }

```

Comment:
```suggestion
```
No need to assign url since you're throwing right after

Reply:
The throw message contains the URL.